lib/redis-connection.js

```diff
@@ -0,0 +1,70 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const error = require('./error')
+const P = require('./promise')
+
+module.exports = (log, client) => {
+  let isUpdating = false
+
+  return {
+    get (key) {
+      return client.getAsync(key)
+        .catch(err => {
+          log.error({ op: 'redis.get.error', key, err: err.message })
+          // Allow callers to distinguish between the null result
+          // and some kind of connection error
+          return false
```

Comment:
I was surprised to see this converted from an exception into a sentinel return value, any particular advantage of taking that approach?  Will other methods like `set` and `del` still treat this case as an error?

Reply:
Only that it saves having the same logic to catch and log the error in two separate places. It's a direct refactoring of what already exists in `db.sessions` and `db.devices`; in both methods, errors from Redis are swallowed because we want the device manager to remain usable but we also need to signal that they shouldn't include `lastAccessTime` in the response.

Explaining it that way does make me realise it's probably the wrong level of abstration to refactor it to though. If we start using Redis for other reasons, I guess it's unlikely that those other methods would have the same requirements. I'll move it to a helper function inside `lib/db` instead.